dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -185,6 +181,15 @@ public Entry addAliases(@ApiParam(hidden = true) @Parameter(hidden = true, name
         return sourceFiles.stream().map(sourceFile -> sourceFile.getType()).collect(Collectors.toCollection(TreeSet::new));
     }
 
+    public void checkEntryPermissions(final Optional<User> user, final Entry<? extends Entry, ? extends Version> entry) {
+        if (!entry.getIsPublished()) {
+            if (user.isEmpty()) {
+                throw new CustomWebApplicationException("This entry is not published.", HttpStatus.SC_UNAUTHORIZED);
```

Comment:
I believe you should return a 404 in this case. If you go to a private GitHub repo that you don't have permissions to, you get a 404, not an unauthorized error.

The logic behind this is that you don't even want to indicate that it exists to an evil-doer. If they know it exists, that's the first step. Then they could then start trying to break into it.

I'll let others opine on this.